### PR TITLE
Linting fixes

### PR DESCRIPTION
**Pull request by turban**
This PR fixes all linting errors in the source code, and adds a lint check to the pre-commit hook.

### CODE REVIEW

File: src/components/map/Layer.js [hunk 1]
```diff
@@ -12,13 +12,16 @@ class Layer extends PureComponent {
...
+        data: PropTypes.array,
+        dataFilters: PropTypes.array,
         id: PropTypes.string.isRequired,
         index: PropTypes.number,
         editCounter: PropTypes.number,
         opacity: PropTypes.number,
         isVisible: PropTypes.bool,
         config: PropTypes.object,
-        // labels: PropTypes.object, // TODO: Is sometimes boolean
+        labels: PropTypes.bool,
```
Comment (amcgee) [inline src/components/map/Layer.js]: Is this always bool or should it be a union type?

File: src/loaders/thematicLoader.js [hunk 2]
```diff
@@ -70,7 +70,7 @@ const thematicLoader = async config => {
         title: name,
         period: period
             ? names[data.metaData.dimensions.pe[0]]
-            : `${formatLocaleDatee(config.startDate)} - ${formatLocaleDate(
+            : `${formatLocaleDate(config.startDate)} - ${formatLocaleDate(
```
Comment (amcgee) [inline src/loaders/thematicLoader.js]: YAY LINTING!

File: src/reducers/map.js [hunk 3]
```diff
@@ -260,10 +265,12 @@ const map = (state = defaultState, action) => {
                 ...state,
             };
 
+        /*
```
Comment (amcgee) [inline src/reducers/map.js]: Should we just delete this case?

File: src/util/sld.js [hunk 4]
```diff
@@ -25,10 +25,12 @@ export const createSld = () => {
 
         let stopFilter = 'PropertyIsLessThan';
 
+        /*
```
Comment (amcgee) [inline src/util/sld.js]: Any reason to keep this commented block?  Why was is safe to remove?

File: src/loaders/thematicLoader.js (hunk 2 above)
Comment (turban) [reply src/loaders/thematicLoader.js]: Need to backport this fix to 2.31. Will do.

File: src/components/map/Layer.js (hunk 1 above)
Comment (amcgee) [reply src/components/map/Layer.js]: What was the resolution here?

Comment (turban) [reply src/components/map/Layer.js]: It is always boolean

File: src/loaders/thematicLoader.js (hunk 2 above)
Comment (turban) [reply src/loaders/thematicLoader.js]: Fix backported to 2.31

**Pull request merged**